garak/resources/beast/beast_attack.py

```diff
@@ -0,0 +1,403 @@
+# SPDX-FileCopyrightText: Portions Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+from typing import Union, Optional, Tuple
+
+import torch
+import numpy as np
+
+import garak._config
+from garak.generators import Generator
+from garak.resources.common import REJECTION_STRINGS, load_advbench
+from tqdm import tqdm
+
+# System prompt pulled from reference implementation
+SYSTEM_PROMPT = (
+    "A chat between a curious user and an artificial intelligence assistant. "
+    "The assistant gives helpful, detailed, and polite answers to the user's questions."
+)
+beast_resource_data = garak._config.transient.basedir / "resources" / "beast" / "data"
+
+
+def _format_chat(generator: Generator, prompt: str):
+    chat = [
+        {"role": "system", "content": SYSTEM_PROMPT},
+        {"role": "user", "content": prompt},
+    ]
+    formatted_prompt = generator.tokenizer.apply_chat_template(chat, tokenize=False)
+    return formatted_prompt
+
+
+@torch.no_grad()
+def _evaluate(generator, prompt, candidate):
+    candidate_str = generator.tokenizer.decode(candidate)
+    input_str = prompt + candidate_str
```

Comment:
We don't add whitespace when generating the suffix, so I'm going to say no? But it may be reasonable to compute the suffix with whitespace and then add it here. 
Whatever we do, it should mirror across both.